app/certificates.go

```diff
@@ -0,0 +1,442 @@
+							Aliases:  []string{"org"},
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:  "organization-unit",
+							Usage: "The name of the organization unit (optional)",
+						},
+						&cli.StringFlag{
+							Name:     "duration",
+							Usage:    "The duration of the client certificate, for example: 1y30d (1 year and 30 days)",
+							Aliases:  []string{"d"},
+							Required: true,
+							Action: func(_ *cli.Context, v string) error {
+								if _, err := utils.ParseDuration(v); err != nil {
+									return fmt.Errorf("failed to parse duration: %v", err)
+								}
+								return nil
+							},
+						},
+						&cli.PathFlag{
+							Name:     CaCertificateFileFlagName,
+							Usage:    "The path of the file where the certificate-authority is stored at",
+							Aliases:  []string{"ca-cert"},
+							Required: true,
+						},
+						&cli.PathFlag{
+							Name:     CaPrivateKeyFileFlagName,
+							Usage:    "The path of the file where the certificate-authority's private key is stored at",
+							Aliases:  []string{"ca-key"},
+							Required: true,
+						},
+						&cli.PathFlag{
+							Name:     "certificate-file",
+							Usage:    "The path of the file to store the generated client certificate in",
+							Aliases:  []string{"cert"},
+							Required: true,
+						},
+						&cli.PathFlag{
+							Name:     "private-key-file",
+							Usage:    "The path of the file to store the generated client private key in",
+							Aliases:  []string{"key"},
+							Required: true,
+						},
+					},
+					Action: func(ctx *cli.Context) error {
+						duration, err := utils.ParseDuration(ctx.String("duration"))
+						if err != nil {
+							return err
+						}
+						caPem, err := ioutil.ReadFile(ctx.Path(CaCertificateFileFlagName))
+						if err != nil {
+							return fmt.Errorf("failed to read ca-cert-file: %s", err)
+						}
+						caPrivKey, err := ioutil.ReadFile(ctx.Path(CaPrivateKeyFileFlagName))
+						if err != nil {
+							return fmt.Errorf("failed to read ca-private-key-file: %s", err)
+						}
+						certPem, certPrivKey, err := generateCertificate(generateCertificateInput{
+							Organization:     ctx.String("organization"),
+							OrganizationUnit: ctx.String("organization-unit"),
+
+							Duration:        duration,
+							CaPem:           caPem,
+							CaPrivateKeyPEM: caPrivKey,
+						})
+						if err != nil {
+							return fmt.Errorf("failed to generate certificate: %s", err)
+						}
+						yes, err := ConfirmPrompt(
+							ctx,
+							fmt.Sprintf("storing the certificate private key at %s, do not share this key with anyone. confirm:",
+								ctx.Path("private-key-file"),
+							),
+						)
+						if err != nil || !yes {
+							return nil
+						}
+						err = ioutil.WriteFile(
+							ctx.Path("certificate-file"),
+							certPem,
+							0600,
+						)
+						if err != nil {
+							return fmt.Errorf("failed to write ca certificate: %s", err)
```

Comment:
error messages here and below refer to 'ca'

Reply:
thank you fixed.